### PR TITLE
docs: Add saucectl docker push cli

### PR DESCRIPTION
**Pull request by tianfeng92**
Add saucectl docker push cli documentation.

### Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

File: docs/dev/cli/saucectl/docker/push.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+## Usage
+
+```bash
+$ saucectl docker push image_name [flags]
+```
+
+:::note
+Ensure your `image_name` adheres to our Container Registry pattern.
+
+In order to join SauceLabs Container Registry, please contact SauceLabs customer support.
```
Comment (alexplischke) [inline docs/dev/cli/saucectl/docker/push.md:20]:
```suggestion
In order to join the Sauce Labs Container Registry, please contact SauceLabs customer support.
```

File: docs/dev/cli/saucectl/docker/push.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+Explore [additional details](/docs/orchestrate/saucelabs-private-registry.md) about the Sauce Labs Container Registry.
+
+## Usage
+
+```bash
+$ saucectl docker push image_name [flags]
+```
+
+:::note
+Ensure your `image_name` adheres to our Container Registry pattern.
```
Comment (mhan83) [inline docs/dev/cli/saucectl/docker/push.md:18]: Is the naming pattern documented somewhere that we can link to here?

Comment (tianfeng92) [reply docs/dev/cli/saucectl/docker/push.md:18]: I don't think there is any documented pattern. The only related documentation is contacting customer service 😂

Comment (mhan83) [reply docs/dev/cli/saucectl/docker/push.md:18]: derp.

Should we just remove this warning then? What's the use in telling the reader that `image_name` has to follow a specific pattern if we're not going to tell them either what the pattern is or what to do if they encounter an error due to how they've named their image.

Comment (tianfeng92) [reply docs/dev/cli/saucectl/docker/push.md:18]: Sounds fair!

**Review by alexplischke — APPROVED** [review]

**Review by mhan83 — APPROVED** [review]

**Pull request merged**